src/static-template.js

```diff
@@ -71,9 +72,14 @@ function html(strings) {
  * @return {!Object<string, !Element>}
  */
 export function htmlRefs(root) {
-  const elements = root.querySelectorAll('[ref]');
+  let elements = root.querySelectorAll('[ref]');
   const refs = map();
 
+  if (root.hasAttribute('ref')) {
```

Comment:
@jridgewell Added this as a convenience. The alternative would be to modify the result of `htmlRefs` when root + refs are meant to be referenced in the same object.

Reply:
Actually, nvm. I'm noticing that it was designed specifically not to set refs on root.